File: apps/api/app/models/persona.py
def serialize_persona(doc: dict) -> dict:
    return {
        "id": doc["_id"],
        "display_name": doc["display_name"],
        "region_label": doc["region_label"],
        "region_lat": doc["region_lat"],
        "region_lng": doc["region_lng"],
        "profile_signal_id": str(doc["profile_signal_id"]) if doc.get("profile_signal_id") else None,
        "is_demo": doc.get("is_demo", True),
    }

File: apps/web/src/lib/persona.ts
export type Persona = {
  id: string;
  display_name: string;
  region_label: string;
  region_lat: number;
  region_lng: number;
  profile_signal_id: string | null;
  is_demo: boolean;
};

const STORAGE_KEY = "sameworld_persona";

export function getStoredPersona(): Persona | null {
  if (typeof window === "undefined") return null;
  const raw = localStorage.getItem(STORAGE_KEY);
  return raw ? (JSON.parse(raw) as Persona) : null;
}

export function storePersona(persona: Persona) {
  localStorage.setItem(STORAGE_KEY, JSON.stringify(persona));
}

export function clearPersona() {
  localStorage.removeItem(STORAGE_KEY);
}
